src/gwc/src/main/java/org/geoserver/gwc/dispatch/GwcServiceDispatcherCallback.java

```diff
@@ -47,6 +48,15 @@
     private static final Pattern GWC_LAYER_VIRTUAL_SERVICE_PATTERN =
             Pattern.compile("([^/]+)/([^/]+)/gwc/service.*");
 
+    static final String buildRestPattern(int numPathElements) {
+        return ".*/service/wmts/rest" + Strings.repeat("/([^/]+)", numPathElements);
+    }
+
+    private static final Pattern TILE_1 = Pattern.compile(buildRestPattern(5));
+    private static final Pattern TILE_2 = Pattern.compile(buildRestPattern(6));
+    private static final Pattern FEATUREINFO_1 = Pattern.compile(buildRestPattern(7));
+    private static final Pattern FEATUREINFO_2 = Pattern.compile(buildRestPattern(8));
+
```

Comment:
We have 1 and 2 to vouch for virtual services ?

Reply:
No it's because the GWC services accept a PATH with or without the style name in the middle, hence it's either a 5 or 6 params path for the tile, and a 7 or 8 params path for the feature info